pkg/intern/intern.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2016 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// A general purpose string interning package to reduce memory
+// consumption and improve processor cache efficiency.
+package intern
+
+import (
+	"sync"
+)
+
+type Pool struct {
+	sync.RWMutex
+	strings  map[string]string
+	maxCount int
```

Comment:
maxCount is one way to limit the pool. would we ever consider a size-defined limit?